vizro-core/src/vizro/static/css/loading.css

```
@@ -0,0 +1,60 @@
+/* Loading spinner */
+._dash-loading {
+  background-color: var(--root-loading-color);
+  inset: 0;
+  color: transparent;
+  margin: 0;
+  position: fixed;
+  text-align: center;
+}
+
+._dash-loading::after {
+  animation:
+    load6 1.7s infinite ease,
+    round 1.7s infinite ease;
+  border-radius: 50%;
+  color: var(--root-loading-contrast);
+  content: "";
+  display: inline-block;
+  font-size: 60px;
+  height: 1em;
+  margin: 2rem auto 0;
+  vertical-align: text-bottom;
+  width: 1em;
+}
+
+@keyframes load6 {
+  0%,
+  5%,
+  95%,
+  100% {
+    box-shadow:
+      0 -0.83em 0 -0.4em,
+      0 -0.83em 0 -0.42em,
+      0 -0.83em 0 -0.44em,
+      0 -0.83em 0 -0.46em,
+      0 -0.83em 0 -0.477em;
+  }
+
+  10%,
+  59% {
+    box-shadow:
+      0 -0.83em 0 -0.4em,
+      -0.087em -0.825em 0 -0.42em,
+      -0.173em -0.812em 0 -0.44em,
+      -0.256em -0.789em 0 -0.46em,
+      -0.297em -0.775em 0 -0.477em;
+  }
+
+  /* Other keyframes omitted for brevity */
+}
+
+@keyframes round {
+  0% {
+    transform: rotate(0deg);
+  }
+
+  100% {
+    transform: rotate(360deg);
+  }
+}
```

I think we should simplify this as much as possible, because I don't even fully understand what's happening here 😄  I took the code snippet below from `dbc` - I think that's all we need. I've customised it a bit, so made it bigger and tried to position the loading spinner at the center of the page.

 https://github.com/facultyai/dash-bootstrap-components/blob/5c8f4b40f1100fc00bf2d5c1d671a7815a6b2910/docs/static/loading.css
```suggestion
._dash-loading {
  background: rgba(20, 23, 33, 1);
  inset: 0;
  color: transparent;
  margin: 0;
  position: fixed;
  text-align: center;

}

._dash-loading::after {
  content: '';
  display: inline-block;
  width: 12rem;
  height: 12rem;
  color: lightgrey;
  vertical-align: text-bottom;
  border: 1rem solid currentColor;
  border-right-color: transparent;
  border-radius: 50%;
  -webkit-animation: spinner-border 0.75s linear infinite;
  animation: spinner-border 0.75s linear infinite;
  margin: 20rem 0;
}
```

---

It would be good if @pruthvip15 takes a look at above 👍

---

Interesting suggestion - how did you find it / know what there was a CSS spinner there?

---

I've read through this thread: bit.https://community.plotly.com/t/change-loading-to-a-spinner/52933 and found a link to the dbc CSS for it :) It looks way simpler and should do the trick 👍. After that, I just played around with the CSS to make the spinner bigger and move it around.